cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -293,6 +303,90 @@ Status DiscoverFilesFromDir(const std::shared_ptr<fs::LocalFileSystem>& local_fs
   return Status::OK();
 }
 
+namespace internal {
+
+ARROW_ENGINE_EXPORT Status ParseAggregateMeasure(
```

Comment:
Is this a refactor?

Reply:
Yes, this is another method-extraction refactoring.